opensearch-operator/pkg/builders/cluster.go

```diff
@@ -372,6 +372,20 @@ func NewSTSForNodePool(
 			SubPath:   "opensearch.keystore",
 		})
 
+		// Add volume and volume mount for config, useful for readOnlyRootFilesystem
+		volumes = append(volumes, corev1.Volume{
+			Name: "config-emptydir",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
+
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
```

Comment:
The mount needs to be for only the init container

Reply:
this should do it.